docs/writing-tests/testharness-api.md

```diff
@@ -430,18 +430,39 @@ not possible to make the test reliable in some other way.
 
 ## Setup ##
 
-Sometimes tests require non-trivial setup that may fail. For this purpose
-there is a `setup()` function, that may be called with one or two arguments.
-The two argument version is:
+Sometimes tests require non-trivial setup that may fail. testharness.js
+provides two global functions for this purpose, `setup` and `promise_setup`.
+
+`setup()` may be called with one or two arguments. The two argument version is:
 
 ```js
 setup(func, properties)
 ```
 
-The one argument versions may omit either argument.
-func is a function to be run synchronously. `setup()` becomes a no-op once
-any tests have returned results. Properties are global properties of the test
-harness. Currently recognised properties are:
+The one argument versions may omit either argument. `func` is a function to be
+run synchronously. `setup()` becomes a no-op once any tests have returned
+results. `properties` is an object which specifies global properties of the
+test harness (enumerated in the following section).
+
+`promise_setup()` allows authors to pause testing until some asynchronous
+operation has completed. It has the following signature:
+
+```js
+promise_setup(func, properties)
+```
+
+Here, the `func` argument is required. The harness will defer execution of this
+function until any subtests have completed. The function must return an object
+with a `then` method (e.g. an ECMAScript Promise instance); the harness will
+wait for the fulfillment of this value before executing any additional tests
```

Comment:
```suggestion
wait for the fulfillment of this value before executing any additional subtests
```

I'm not sure if this suggestion is correct; this document seems a bit inconsistent in using 'test' vs 'subtest' (the entire document, not just your edits)

Reply:
It is inconsistent. Since we're not going to solve that in this patch, I'd
rather side on the more precise term as you've suggested.